components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/DateRangePicker.jsx

```diff
@@ -211,40 +219,35 @@ export default class DateRangePicker extends Widget {
     }
 
     loadDefaultTimeRange() {
-        if (location.search !== '') {
-            let dateTimeRangeInfo = JSON.parse(decodeURI(location.search.substr(1)));
-            if (dateTimeRangeInfo.hasOwnProperty('tr')) {
-                if (dateTimeRangeInfo.tr.toLowerCase() === 'custom') {
-                    if (dateTimeRangeInfo.hasOwnProperty('sd')
-                        && dateTimeRangeInfo.hasOwnProperty('ed')) {
-                        if (dateTimeRangeInfo.hasOwnProperty('g')) {
-                            this.loadUserSpecifiedCustomTimeRange(dateTimeRangeInfo.sd,
-                                dateTimeRangeInfo.ed, dateTimeRangeInfo.g)
-                        } else {
-                            this.loadUserSpecifiedCustomTimeRange(dateTimeRangeInfo.sd,
-                                dateTimeRangeInfo.ed, '')
-                        }
-                    } else {
-                        this.handleGranularityChange(this.getDefaultTimeRange());
-                    }
-                } else {
-                    if (dateTimeRangeInfo.hasOwnProperty('sync')) {
-                        if (dateTimeRangeInfo.sync === true) {
-                            this.setState({
-                                enableSync: true,
-                                btnType: <NotificationSync color='#f17b31'/>
-                            });
-                        }
-                    }
+        let dateTimeRangeInfo = this.getDateTimeRangeInfo();
+        if (dateTimeRangeInfo.hasOwnProperty('tr')) {
```

Comment:
Why `hasOwnProperty`?

Reply:
Changed to using '.property' instead of hasOwnProperty and merged in PR #63